src/ServiceControl.AcceptanceTests/HeartbeatMonitoring/When_an_unmonitored_endpoint_is_marked_as_monitored.cs

```diff
@@ -0,0 +1,105 @@
+﻿namespace ServiceBus.Management.AcceptanceTests.HeartbeatMonitoring
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Contexts;
+    using NServiceBus;
+    using NServiceBus.AcceptanceTesting;
+    using NServiceBus.Features;
+    using NUnit.Framework;
+    using ServiceControl.CompositeViews.Endpoints;
+    using Conventions = NServiceBus.AcceptanceTesting.Customization.Conventions;
+
+    public class When_an_unmonitored_endpoint_is_marked_as_monitored : AcceptanceTest
+    {
+        enum State
+        {
+            WaitingForEndpointDetection,
+            WaitingForHeartbeatFailure
+        }
+
+        static string EndpointName => Conventions.EndpointNamingConvention(typeof(MyEndpoint));
+
+        [Test]
+        public void It_is_shown_as_inactive_if_it_does_not_send_heartbeats()
+        {
+            var context = new MyContext();
+            List<EndpointsView> endpoints = null;
+            var state = State.WaitingForEndpointDetection;
+            Define(context)
+                .WithEndpoint<MyEndpoint>()
+                .Done(c =>
+                {
+                    if (state == State.WaitingForEndpointDetection)
+                    {
+                        var found = TryGetMany("/api/endpoints/", out endpoints, e => e.Name == EndpointName && !e.Monitored);
+                        if (found)
+                        {
+                            var endpointId = endpoints.First(e => e.Name == EndpointName).Id;
+                            Patch($"/api/endpoints/{endpointId}",new EndpointUpdateModel
+                            {
+                                MonitorHeartbeat = true
```

Comment:
@johnsimons well, an alternative would be to not allow to enable monitoring of an endpoint if we haven't received any HBs. I think I prefer this approach. Here, by force-enabling monitoring we say "I expect this endpoint is going to have HB plugin installed and working" and it will show in read until somebody actually installs and configures it.